lib/chat/query.py

```diff
@@ -0,0 +1,44 @@
+import graphene
+
+from bson import ObjectId
+
+from .model import Chat
+from .type import ChatType
+
+from lib.chatMessage.model import ChatMessage
+from lib.match.model import Match
+from lib.adoptable.model import Adoptable
+
+
+class Query:
+    chat = graphene.Field(ChatType, adoptable_id=graphene.String())
+
+    def resolve_chat(self, info, adoptable_id):
+        token = getattr(info.context, "token", None)
+
+        if token is None:
+            return {}
+
+        chat = Chat.objects(adoptable_id=ObjectId(adoptable_id)).all()
+
+        if len(chat) != 1:
+            return {}
+
+        chat = chat[0]
```

Comment:
why don't you do this in the chat = Chat.objects().all() statement?

Reply:
@BeauTaapken The chat might actually not exist, in the current version it is not a guarantee a chat exists